src/sinks/influxdb/logs.rs

```diff
@@ -0,0 +1,657 @@
+
+        let tag_keys: Vec<String> = [
+            vec!["host".to_owned(), "source_type".to_owned()],
+            self.tags.clone(),
+        ]
+        .concat();
+
+        // Measurement
+        let measurement = encode_namespace(&self.namespace, &"vector");
+
+        // Timestamp
+        let timestamp =
+            if let Some(Value::Timestamp(ts)) = event.remove(log_schema().timestamp_key()) {
+                encode_timestamp(Some(ts))
+            } else {
+                encode_timestamp(None)
+            };
+
+        // Tags
+        let tags: BTreeMap<String, String> = event
+            .all_fields()
+            .filter(|(key, _)| tag_keys.contains(key))
+            .map(|(key, value)| (key, value.to_string_lossy()))
+            .collect();
+
+        // Fields
+        let fields: HashMap<String, Field> = event
+            .all_fields()
+            .filter(|(key, _)| !tag_keys.contains(key))
+            .map(|(key, value)| (key, value.to_field()))
+            .collect();
+
+        influx_line_protocol(
+            measurement,
+            "logs",
+            Some(tags),
+            Some(fields),
+            timestamp,
+            &mut output,
+        );
+
+        Some(output.into_bytes())
+    }
+
+    fn build_request(&self, events: Self::Output) -> http::Request<Vec<u8>> {
+        let settings = influxdb_settings(
+            self.influxdb1_settings.clone(),
+            self.influxdb2_settings.clone(),
+        )
+        .unwrap();
+
+        let endpoint = self.endpoint.clone();
+        let token = settings.token();
+
+        let uri = settings.write_uri(endpoint).unwrap();
+
+        let mut builder = Request::builder();
+        builder.method(Method::POST);
+        builder.uri(uri.clone());
+
+        builder.header("Content-Type", "text/plain");
+        builder.header("Authorization", format!("Token {}", token));
+        builder.body(events).unwrap()
+    }
+}
+
+impl Value {
+    pub fn to_field(&self) -> Field {
+        match self {
+            Value::Integer(num) => Field::Int(*num),
+            Value::Float(num) => Field::Float(*num),
+            Value::Boolean(b) => Field::Bool(*b),
+            _ => Field::String(self.to_string_lossy()),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::event::Event;
+    use crate::sinks::influxdb::test_util::{assert_fields, split_line_protocol, ts};
+    use crate::sinks::util::http::HttpSink;
+    use crate::sinks::util::test::build_test_server;
+    use crate::test_util;
+    use chrono::offset::TimeZone;
+    use chrono::Utc;
+    use futures01::{Sink, Stream};
+
+    #[test]
+    fn test_config_without_tags() {
+        let config = r#"
+            namespace = "vector-logs"
+            endpoint = "http://localhost:9999"
+            bucket = "my-bucket"
+            org = "my-org"
+            token = "my-token"
+        "#;
+
+        let _: InfluxDBLogsConfig = toml::from_str(&config).unwrap();
```

Comment:
If you want to avoid the assignment, you can write this as:
```rust
toml::from_str::<InfluxDBLogsConfig>(&config).unwrap();
```

Reply:
Done